analyses/rna-seq-composition/rna-seq-composition.Rmd

```diff
@@ -0,0 +1,583 @@
+  mutate(ref_range_name = factor(ref_range_name, levels = ref_range_codes$ref_range_name),
+         within_limits = factor(within_limits, levels = limit_status_levels))
+
+```
+
+
+## Tabulate excluded samples
+```{r}
+
+flag_summary_stats <- read_fracs_with_thresholds_anno  %>%
+  mutate(`Fraction of sample` = gsub("\n", "", stat_label[match(ref_range_name, names(stat_label))])) %>%
+  #  group_by(ref_range_name) %>%
+  tabyl(`Fraction of sample`, within_limits) %>%
+  adorn_totals(where = c("col")) %>%
+  adorn_percentages(denominator = "row") %>% 
+  adorn_pct_formatting(digits = 0) %>%
+  adorn_ns() %>%
+  mutate(`Fraction of sample` = factor(`Fraction of sample`, 
+                                       levels = gsub("\n", "", stat_label))) %>%
+  arrange(`Fraction of sample`)
+```
+
+
+## Histogram portion of plot
+```{r}
+
+plot_title <- paste0(sum(threshold_flag_annotations$flagged_thresholds!=""), " samples are outside reference ranges")
+
+p20 <-  ggplot(read_fracs_with_thresholds_anno) + 
+  geom_histogram(aes(x = sample_value, fill = ref_range_name)) + 
+  geom_vline(data=all_stats,
+             aes(xintercept = value, linetype = stat, color = ref_range_name)) +
+    scale_fill_viridis_d() +
+  scale_color_viridis_d() +
+  facet_wrap(~ref_range_name, nrow=1, labeller = labeller(
+    ref_range_name=stat_label) )
+
+p20b <- p20 + 
+  theme(legend.position = "none") +
+  ylab("Samples") + 
+  xlab(paste0("Relative sample composition, n=", length(unique(read_fracs_with_thresholds$sample_id)))) 
+
+padding_from_line <- -0.04
+
+read_type_frac_thresholds_for_fig2 <- all_stats %>%
+  filter(grepl("two_sd", stat_name)) %>%
+  mutate(
+    x = ifelse(stat_name=="mean_plus_two_sd", value + padding_from_line, value - padding_from_line),
+    y = ifelse(stat_name=="mean_plus_two_sd", 200,300),
+    label = ifelse(stat_name=="mean_plus_two_sd",
+                   paste0("mean\n+ 2sd\n", round(value, 3)),
+                   paste0("mean\n- 2sd\n", round(value, 3))
+                   )
+    )
+
+p20c <- 
+  p20b + 
+  scale_x_continuous(breaks = seq(0, 1, by=0.25)) +
+  expand_limits(x = c(-0.25, 0.9)) +
+  ggtitle(plot_title)
+
+p20d <- 
+  p20c + 
+  geom_label(data = read_type_frac_thresholds_for_fig2, 
+            aes(x = x, y = y, label = round(value, 3)),#, hjust = hjust)
+  ) 
+
+```
+
+# Generate summary plot
+```{r}
+
+tbl <- tableGrob(flag_summary_stats, rows=NULL)
+
+wrap_plots(p20c, tbl, ncol=1, heights = 2:1)
+
+ggsave(file.path(plots_dir, "fractions_and_reference_ranges.png"), width = 8, height = 6)
+
+```
+
+# Export table of flagged samples
+```{r}
+
+samples_to_export <- c(read_fracs_with_thresholds_anno %>%
+  filter(flagged_thresholds !="") %>%
+  pull(sample_id) %>%
+  unique, 
+  depth_status %>%
+    filter(status == "<10M_MEND_reads") %>%
+    pull(sample_id))
+  
+read_fracs_with_thresholds_anno %>%
+  filter(sample_id %in% samples_to_export) %>%
+  mutate(within_limits = gsub("within_limits", "", within_limits)) %>%
```

Comment:
I see your point. I agree that it could be valuable for all samples to be included. Do you think there should be an additional column called summary_flag_status that contains either a summary flag code (like "NM, MM, D") or "not flagged"?

Reply:
(I added the column)